compiler/rustc_transmute/src/maybe_transmutable/mod.rs

```diff
@@ -75,12 +75,10 @@ mod rustc {
                 let dst = Tree::from_ty(dst, context);
 
                 match (src, dst) {
-                    // Answer `Yes` here, because 'unknown layout' and type errors will already
-                    // be reported by rustc. No need to spam the user with more errors.
-                    (Err(Err::TypeError(_)), _) => Err(Answer::Yes),
-                    (_, Err(Err::TypeError(_))) => Err(Answer::Yes),
-                    (Err(Err::Unknown), _) => Err(Answer::Yes),
-                    (_, Err(Err::Unknown)) => Err(Answer::Yes),
```

Comment:
Why not just return `No` here? If this is just to not annoy users with errors, we should be doing diagnostic deduplication on the error reporting side.

Reply:
Yeah, I added deduplication here

https://github.com/rust-lang/rust/blob/0d74752a385285d87b14c88cbc7eacdf11766dea/compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs#L743-L754

Is that what you mean?

The reason I made a separate `Answer` variant instead of reusing `Answer::No` is cause there is a semantic difference between safe transmute not being possible (`Answer::No`) and not knowing if safe transmute is possible because of an error that occurred  when trying to get an answer (`Answer::Err`).

Reply:
Well in the case of a layout error, it seems like you should be modeling this case with an additional variant that signifies "unknown".